src/ui_basic/panel.cc

```diff
@@ -571,10 +571,19 @@ void Panel::think() {
  * (grand-)children for which set_thinks(false) has not been called.
  */
 void Panel::do_think() {
+	if (is_dying()) {
+		return;
+	}
+
 	if (thinks()) {
 		think();
 	}
 
+	// think() may have called die()
```

Comment:
Why doing this twice? could we not just wait another think() cycle?
If not what about this code?
if (!is_dying() && thinks() ) {
	return;
}